### ISSUE TITLE
Not working with npm workspaces?

### ISSUE THREAD

**Issue by comfortliner** [thenativeweb/roboter#807]
It seems not to work in combination with workspaces.
Snippet from `<projectroot>/package.json`:
```json
  "workspaces": [
    "packages/*"
  ]
```
`roboter` is installed in the `<projectroot>/node_modules`, so this works in project root folder:
```shell
npx roboter --help
```
Running the following command in the same folder:
```shell
npm exec --workspace=packages/backend -- roboter --help
```
throws an error:
```shell
node:internal/errors:464
    ErrorCaptureStackTrace(err);
    ^

TypeError [ERR_INVALID_FILE_URL_HOST]: File URL host must be "localhost" or empty on darwin
    at new NodeError (node:internal/errors:371:5)
    at getPathFromURLPosix (node:internal/url:1397:11)
    at fileURLToPath (node:internal/url:1420:50)
    at finalizeResolution (node:internal/modules/esm/resolve:395:16)
    at moduleResolve (node:internal/modules/esm/resolve:932:10)
    at defaultResolve (node:internal/modules/esm/resolve:1044:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:422:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:222:40)
    at ESMLoader.import (node:internal/modules/esm/loader:276:22)
    at initializeLoader (node:internal/process/esm_loader:74:49) {
  code: 'ERR_INVALID_FILE_URL_HOST'
}
```

**Comment by goloroden** [comment]
Hi @comfortliner 👋 

Thanks for bringing up this issue.

Generally speaking, of course we would like to help, on the other hand we don't use npm workspaces at all, and we don't want to put efforts into this (especially in the long run this means maintaining things we don't use).

Anyway, we are going to have a look at this, to figure out whether it's a small thing to fix, or whether it requires larger changes and extensions to make things work. In the former case, we'll fix it, in the latter we won't. Either way, we'll let you know what we figured out.

Have a nice day,


Golo

**Comment by yeldiRium** [comment]
I have found the cause of the error: When starting the roboter we try to find the `ts-node` package to be able to run the roboter, which is written using ES modules. This search for the ts-node package fails in the case of a workspace, since the `ts-node` package is installed in an unexpected relative location.

This is simple enough to fix. But the problems don't end there.

Since the roboter bundles quite a few dependencies (nowadays as auto-installed peerDependencies), we rely on finding the locations of installed npm packages in quite a few places. This mainly concerns the various linter libraries, plugins and rules.

The workspaces mess up all that package-finding logic since `npx exec` runs the roboter in the context of the workspace, where none of these dependencies are available, since they are installed on the project's root level. Fixing this would require at least a few changes to the roboter. I actually think that these changes would make the roboter more robust by relying more on node.js' module resolution instead of trying to find packages ourselves via `npm ls` or similar methods.

Originally I wanted to continue this comment saying that I thought the changes needed were too extensive to be worth it. But the issue was annoying me too much to let it go - and I found a solution so simple it annoys me even more than the actual issue.

I've had at least three different approaches to locating dependencies in the roboter over the last two years. I never thought to actually use `require.resolve` to leverage node.js' own module resolution strategy - which is __obviously__ what we want.

Switching the current implementation of the package finding algorithm to just use `require.resolve` has solved all these issues and makes the roboter usable with workspaces. Thank you for your time.

@goloroden PR incoming.
